WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -217,13 +222,46 @@ public void onSiteRemoved(OnSiteRemoved event) {
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onSiteChanged(OnSiteChanged event) {
-        mDebouncer.debounce(Void.class, new Runnable() {
-            @Override public void run() {
-                if (!isFinishing()) {
-                    getAdapter().loadSites();
-                }
+        if (mSwipeToRefreshHelper.isRefreshing()) {
+            mSwipeToRefreshHelper.setRefreshing(false);
+            if (!isFinishing()) {
+                getAdapter().loadSites();
             }
-        }, 200, TimeUnit.MILLISECONDS);
+        } else {
+            mDebouncer.debounce(Void.class, new Runnable() {
+                @Override
+                public void run() {
+                    if (!isFinishing()) {
+                        getAdapter().loadSites();
+                    }
+                }
+            }, 200, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void initSwipeToRefreshHelper(View view) {
+        if (view == null) {
+            return;
+        }
+        mSwipeToRefreshHelper = new SwipeToRefreshHelper(
+                this,
+                (CustomSwipeRefreshLayout) view.findViewById(R.id.ptr_layout),
+                new SwipeToRefreshHelper.RefreshListener() {
+                    @Override
+                    public void onRefreshStarted() {
+                        if (isFinishing()) {
+                            return;
+                        }
+                        if (!NetworkUtils.checkConnection(SitePickerActivity.this)) {
+                            mSwipeToRefreshHelper.setRefreshing(false);
+                            ToastUtils.showToast(SitePickerActivity.this, R.string.site_picker_no_network_refresh_error);
+                            return;
+                        }
+                        mDispatcher.dispatch(SiteActionBuilder.newRemoveWpcomAndJetpackSitesAction());
```

Comment:
Hey @tonyr59h , got to say, this all-sites removal makes me a bit uncomfortable. I assume the reason you added it is to "clean up" any remotely-deleted sites, right? Is there some reason beyond that?

Here's the thing though: an unadvertised removal like this can possibly delete data not synced to the backend yet, like draft posts that have local changes. Fixing/catering for a case like that might easily fall outside the scope of this PR so, may I suggest instead to avoid removing local sites en-mass? Perhaps we can remove local sites after we get the updated remote list and check that the site doesn't have any local drafts. WDYT?